content/blog/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages.md

```diff
@@ -0,0 +1,263 @@
+---
+title: "Multi-stage YAML pipeline to create and publish NuGet packages"
+date: 2019-08-26T00:00:00+01:00
+publishdate: 2019-08-26T00:00:00+01:00
+lastmod: 2019-08-26T00:00:00+01:00
+image: "images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages.jpg"
+tags: [ "Azure Pipelines", "Azure DevOps", "NuGet", "Continuous Integration", "YAML" ]
+comments: false
+draft: true
+---
+
+I've recently created a NuGet package that I published on nuget.org. To make this process as smoothed and simple as possible I've created a multi-stage yaml pipeline in Azure DevOps to create and publish my package.
+
+My preferred way of working with NuGet packages is to first create a prerelease version of a package and use it in my software. If I'm confident that it works I create a release version.
+
+I've created a single pipeline to support this workflow, using the Multi-stage pipelines feature that's currently in preview. The pipeline:
+
+- builds my solution
+- creates a prerelease version of my package
+- creates a release version of my package
+- automatically publishes the prerelease package to a private Azure DevOps Artifacts feed
+- publishes the release version of the package to nuget.org if I give an approval
+
+In this post I'll go into how the pipeline is set up.
+
+First step is to enable the _Multi-stage pipelines_ preview feature:
+
+- Login to your Azure DevOps environment.
+- Click on your avatar in the top right corner.
+- Choose _Preview features_.
+- Enable the _Multi-stage pipelines_ option.  
+![Multi-stage pipelines preview feature](../../static/images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png)
+<!-- ![Multi-stage pipelines preview feature](../../../../../images/multi-stage-yaml-pipeline-to-create-and-publish-nuget-packages/preview-feature-multi-stage-pipelines.png) -->
+
+Now that the preview feature is enabled we can start. So, create a new pipeline and connect it to one of the sources that supports YAML, like Azure Repositories or GitHub. You can choose any pipeline template, because we're going to clear it and start from scratch. The [Build, test, and deploy .NET Core apps](https://docs.microsoft.com/azure/devops/pipelines/languages/dotnet-core) documentation is a good starting point if you haven't created a pipeline before.
```

Comment:
Als we het toch gaan legen, kunnen we dan niet beter kiezen voor een empty template?

Reply:
Ik zie bij mij geen optie om met een lege pipeline te starten. De 'starter pipeline' komt dan nog het dichtst in de buurt.